ts/IterableInbox.tsx

```diff
@@ -1,30 +1,383 @@
+              percentage: 0
+           },
+           right: {
+              percentage: 0
+           },
+           bottom: {
+              percentage: 0
+           },
+           left: {
+              percentage: 0
+           }
+        }
+     },
+       saveToInbox: true,
+       inboxMetadata: {
+          title: "Dog Moms Unite!",
+          subtitle: "Show up at the dog park",
+          icon: "icon5.png"
+       },
+       customPayload: {},
+       read: false,
+       priorityLevel: 400.5,
+       last: false
+    },
+    {
+       campaignId: 5,
+       messageId: 9,
+       trigger: "",
+       createdAt: "2021-07-21 12:20:00",
+       expiresAt: "2021-07-21 12:20:00",
+       content: {
+        html: "<html><head></head><body>Test</body></html>",
+        inAppDisplaySettings: {
+           top: {
+              percentage: 0
+           },
+           right: {
+              percentage: 0
+           },
+           bottom: {
+              percentage: 0
+           },
+           left: {
+              percentage: 0
+           }
+        }
+     },
+       saveToInbox: true,
+       inboxMetadata: {
+          title: "Dog Moms Unite!",
+          subtitle: "Show up at the dog park",
+          icon: "icon5.png"
+       },
+       customPayload: {},
+       read: false,
+       priorityLevel: 400.5,
+       last: false
+    },
+    {
+       campaignId: 5,
+       messageId: 10,
+       trigger: "",
+       createdAt: "2021-07-21 12:20:00",
+       expiresAt: "2021-07-21 12:20:00",
+       content: {
+        html: "<html><head></head><body>Test</body></html>",
+        inAppDisplaySettings: {
+           top: {
+              percentage: 0
+           },
+           right: {
+              percentage: 0
+           },
+           bottom: {
+              percentage: 0
+           },
+           left: {
+              percentage: 0
+           }
+        }
+     },
+       saveToInbox: true,
+       inboxMetadata: {
+          title: "Dog Moms Unite!",
+          subtitle: "Show up at the dog park",
+          icon: "icon5.png"
+       },
+       customPayload: {},
+       read: false,
+       priorityLevel: 400.5,
+       last: true
+    }
+]
+
 const IterableInbox = () => {
    const inboxTitle = "Inbox"
    const [isDisplayMessage, setIsDisplayMessage] = useState(false)
-   const [selectedMessageIdx, setSelectedMessageIdx] = useState(0)
+   const [selectedMessageId, setSelectedMessageId] = useState(0)
    const [messages, setMessages] = useState(sampleMessages)
 
-   let selectedMessage = messages[selectedMessageIdx]
+   const selectedMessage = messages.find(message => message.messageId === selectedMessageId)
 
-   function handleMessageSelect(index: number, messages: Array<any>) {
-      const newMessages = messages.map((message, messageIndex) => {
-         return (messageIndex === index) ?
+   function handleMessageSelect(id: number, messages: Message[]) {
+      let newMessages = sampleMessages.map((message) => {
```

Comment:
why is this line now using `sampleMessages` rather than the `messages` you're providing as a prop for the function?

Reply:
I agree. The function should be pulling from the state. I made the update.